src/ssl_stubs.c

```diff
@@ -610,7 +616,16 @@ CAMLprim value ocaml_ssl_ctx_set_verify(value context, value vmode, value vcallb
   }
 
   if (Is_block(vcallback))
+  {
+#ifdef NO_NAKED_POINTERS
+    vcallback = Field(vcallback, 0);
+    if (!Is_block(vcallback) || Wosize_val(vcallback) != 1)
```

Comment:
It looks like this section of code previously ignored "failures" if `!Is_block(vcallback)`.

I believe we're introducing a new failure mode in this change. I'm OK with it but it might need to be documented in the interface file. What do you think?

Reply:
`vcallback` has type `verify_callback option`. Previously, `verify_callback` was a naked pointer, so there was no failure mode. This guard is just here in case there really is some C code out there which has a custom verify function which it casts to a naked pointer rather than boxing. If that happens, the program is going to crash - all this guard does is try to display a better error if the supplied doesn't look like the correct kind of block (I guess it could also test the tag).

I think it would be more important to document how to create verify callbacks in the first place before addressing this failure mode - the only way to write one at the moment is to read to the source 😀

Reply:
I just pushed a revised version with the extra test on the `Tag_val` - that means that only the four code pointers `0x4fb`-`0x7fb` would now slip through (which is handy, because those are highly unlikely function addresses)

Reply:
@dra27 to help me understand, where do those four code pointers come from?